### DISCUSSION TITLE
How can I have an array of objects inside Prisma scheme field?

### DISCUSSION THREAD

**Discussion by M4TY** [prisma/prisma#16633] [Q&A]
Hello, this is my current scheme:
```
model Progression {
  id              String @id @default(auto()) @map("_id") @db.ObjectId
  user            User   @relation(fields: [userId], references: [id])
  userId          String
  pack            Pack   @relation(fields: [packId], references: [id])
  packId          String
  problemStatuses String[]
}
```

in problemStatuses, I would like to have array of ProblemStatus:

```
type ProblemStatus = {
    problemId: string,
    status: string,
}
```

Am I able to put array of objects, or alternatively array of type ProblemStatus? I am using MongoDB.

**Comment by nurul3101** [comment]
Hello @M4TY 👋 
You can define your schema like this to get an array of ProblemStatus type
```prisma
model Progression {
  id              String          @id @default(auto()) @map("_id") @db.ObjectId
  user            User            @relation(fields: [userId], references: [id])
  userId          String
  pack            Pack            @relation(fields: [packId], references: [id])
  packId          String
  problemStatuses ProblemStatus[]
}

type ProblemStatus {
  problemId String
  status    String
}
```
Here's a reference to composite types [docs](https://www.prisma.io/docs/concepts/components/prisma-client/composite-types).

**Reply by vladi-stevanovic** [reply]
👋 Hi @M4TY , did you perhaps have a chance to check Nurul's recommendation? Let us know if you still have any questions! 

Or, if you've found a solution, we'd love to know how you solved this problem. Feel free to post it in this thread and "Mark as answer" if this discussion is now solved!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
